File: Modules/SMES/Common/SIE.Web.Andon/Andons/Commands/AndonMessageAddCommand.js
SIE.defineCommand('SIE.Web.Andon.Andons.Commands.AndonMessageAddCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "icon-AddEntity icon-green" },
    onItemCreated: function (entity) {
        var model = entity.data;
        var me = this;
        this.view.execute({
            data: model,
            success: function (res) {
                var data = res.Result;
                entity.setPushPlugId_Display(data.PushPlugName);
                entity.setPushPlugId(data.PushPlugId);
            }
        }, me.view);
    }
});

File: Modules/SMES/Common/SIE.Web.Andon/Andons/Commands/AndonMessageAddCommand.cs
using SIE.Andon.Andons;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.Andon.Andons.Commands
{
    public class AndonMessageAddCommand : ViewCommand
    {
        protected override object Excute(ViewArgs args, string scope)
        {
            var andonMessage = args.Data.ToJsonObject<AndonMessageSend>();
            var andonPushPlugConfig = RT.Service.Resolve<AndonTypeController>().GetAndonConfigPushPlug();
            if(andonPushPlugConfig != null)
            {
                andonMessage.PushPlug = andonPushPlugConfig;
                andonMessage.PushPlugName = andonMessage.PushPlug.Name;
            }
            return andonMessage;
        }
    }
}
